prometheus_api_client/prometheus_connect.py

```diff
@@ -349,3 +352,68 @@ def custom_query_range(self, query: str, start_time: datetime, end_time: datetim
             )
 
         return data
+
+    @retry(stop_max_attempt_number=MAX_REQUEST_RETRIES, wait_fixed=CONNECTION_RETRY_WAIT_TIME)
+    def get_metric_aggregation(self, query, operations):
+        """
+        A method to get aggregations on metric values received from PromQL query.
+
+        This method takes as input a string which will be sent as a query to
+        the specified Prometheus Host. This query is a PromQL query. And, a
+        list of operations to perform such as- sum, max, min, deviation, etc.
+
+        The received query is passed to the custom_query method which returns
+        the result of the query and the values are extracted from the result.
+
+        :param query: (str) This is a PromQL query, a few examples can be found
+        at https://prometheus.io/docs/prometheus/latest/querying/examples/
+        :param operations: (list) A list of operations to perform on the values.
+        Operations are specified in string type.
+        Available operations - sum, max, min, variance, nth percentile, deviation
+        and average.
+
+        :returns: (dict) A dict of aggregated values received in response to the operations
+        performed on the values for the query sent.
+
+        Example output :
+            {
+                'sum': 18.05674,
+                'max': 6.009373
+             }
+        """
+        if not isinstance(operations, list):
+            raise TypeError("Operations can be only of type list")
+        if len(operations) == 0:
+            _LOGGER.debug("No operations found to perform")
+            return None
+        aggregated_values = {}
+        data = self.custom_query(query)
```

Comment:
@4n4nd @amitsagtani97  do you guys think it might be useful to add `params` dict as an optional argument to `get_metric_aggregation`, so that it could then be passed to the `custom_query` call here?

I was trying to test `get_metric_aggregation` with some past data, but realized I can't do that. Although, if this function is meant to perform aggregations on current timestamp values only, then I think this should be fine.

Reply:
+1

Reply:
@chauhankaranraj the aim is to make this function usable for every possible cases. Updated the function with param argument. :+1:

Reply:
Awesome, thanks. Looks good to me